FormFrameworkREADME.md

```diff
@@ -0,0 +1,292 @@
+
+  7. (**Optional**). Specify the `displayLabel:` (within the `displayLayout:` configuration) for the schema as a whole. If the schema is for a code table, the `displayLabel:` describes how the code table entries will be displayed in the dropdown. If the schema is for a record, the `displayLabel:` describes how the record can be succinctly described as a "preview" (of sorts).
+
+  - Example: in "efficacyCodes.schema.yaml" (the schema for a code table), we use the following code snippet to indicate that the value for the 'efficacyRating' field in the schema should be listed as a dropdown menu item for each entry of the code table.
+
+  ```yaml
+    displayLayout:
+      displayLabel: '#(efficacyRating)'  
+      header:  
+        key: efficacyRating.header  
+        default: 'Efficacy %'
+  ```    
+
+  - Example: in "observation.schema.yaml" (the schema for a record), we use the following code snippet to indicate that the date and commonName of the species should be used to describe the observation record when it is displayed in the Inventory page of all recorded observations (as an example). The commonName is a field attached to the species, which is a foreign key reference in the observation schema.
+
+  ```yaml
+    displayLayout:
+      displayLabel: '#(species.commonName) - #(date)'
+  ```
+
+  8. (**Optional**). The `relations:` section is used to link multiple schema records together (when necessary).
+    - Example: there may be 0 to many mechanical treatments associated with an individual observation. In "observation.schema.yaml", the following code snippet indicates that when 1 or more mechanical treatments refer to a given observation as a foreign key, the linked mechanical treatments should be displayed on the front-end side of the observation record.
+
+  ```yaml
+    relations:
+      mechanicalTreatments:
+        header:
+          key: 'observation.mechanicalTreatments'
+          default: Mechanical Treatments
+        description:
+          key: 'observation.mechanicalTreatments.description'
+          default: 'Mechanical treatments performed on observation'
+        type: array
+        relationshipType: one-to-many
+        schema: MechanicalTreatmentSchema
+        meta:
+          embedded: true
+  ```
+
+  9. (**Optional**). If the current schema references any other schemas (e.g., as foreign key references), there are two ways to indicate this in the .yaml file for the current schema:
+
+  - `includes:` specifies the dependent schema's .yaml filename
+  - `externalTables:` specifies the name of the dependent schema's table in the database  
```

Comment:
If dependent/ foreign schema/table definitions are not included in the schema file or any include files, such schema details should be included in externalTables section for schema integrity checking.